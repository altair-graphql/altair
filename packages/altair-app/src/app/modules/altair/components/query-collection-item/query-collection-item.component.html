<div class="query-collections__item">
  <div class="query-collections__item-details-wrapper">
    <div
      class="query-collections__item-details"
      data-test-id="collection-collapse"
      (click)="toggleContent()"
    >
      <div class="query-collections__item-title">
        <span class="query-title-icon" *ngIf="showContent"><app-icon name="folder-minus"></app-icon></span>
        <span class="query-title-icon" *ngIf="!showContent"><app-icon name="folder-plus"></app-icon></span>
        <span>{{ collectionTree.title }}</span></div>
      <div class="query-collections__item-meta">{{ 'COLLECTIONS_NUMBER_OF_REQUESTS' | translate }} {{ getQueryCount(collectionTree) }}</div>
      <!-- <div class="query-collections__item-description">Lorem ipsum dolor sit amet consectetur adipisicing elit. Commodi velit voluptate, delectus possimus corrupti adipisci iusto itaque cum dignissimos sed quae eaque magni harum ratione laborum similique excepturi voluptatibus molestiae?</div> -->
    </div>
    <span
      *ngIf="loggedIn && collectionTree.storageType === 'firestore'"
      class="query-collections__item-synced-icon"
      [title]="'CLOUD_SYNC_TEXT' | translate"
    >
      <app-icon name="cloud"></app-icon>
    </span>
    <div class="query-collections__item-right">
      <span class="query-collections__item-icon">
        <button
          nz-dropdown
          nzTrigger="click"
          [nzDropdownMenu]="collectionItemMenu"
          type="button"
          class="query-collections__item-icon__button"
        >
          <app-icon name="more-horizontal"></app-icon>
        </button>
      </span>
      <nz-dropdown-menu #collectionItemMenu="nzDropdownMenu">
        <ul nz-menu>
          <li
            *ngIf="loggedIn"
            nz-menu-item
            data-test-id="sync-collection"
            (click)="syncCollection()"
          >
            {{ 'COLLECTIONS_MENU_SYNC' | translate }}
          </li>
          <li
            nz-menu-item
            data-test-id="edit-collection"
            (click)="editCollection()"
          >
            {{ 'COLLECTIONS_MENU_EDIT' | translate }}
          </li>
          <li
            nz-menu-item
            data-test-id="delete-collection"
            (click)="deleteCollection()"
          >
            {{ 'COLLECTIONS_MENU_DELETE' | translate }}
          </li>
          <li
            nz-menu-item
            data-test-id="export-collection"
            (click)="exportCollection()"
          >
            {{ 'COLLECTIONS_MENU_EXPORT' | translate }}
          </li>
          <li nz-submenu [nzTitle]="'COLLECTIONS_SORT_BY' | translate">
            <ul>
              <li
                nz-menu-item
                data-test-id="sort-queries-a-z"
                (click)="setQueriesSortBy('a-z')"
              >
                {{ 'COLLECTIONS_SORT_BY_A_Z' | translate }}
              </li>
              <li
                nz-menu-item
                data-test-id="sort-queries-z-a"
                (click)="setQueriesSortBy('z-a')"
              >
                {{ 'COLLECTIONS_SORT_BY_Z_A' | translate }}
              </li>
              <li
                nz-menu-item
                data-test-id="sort-queries-newest"
                (click)="setQueriesSortBy('newest')"
              >
                {{ 'COLLECTIONS_SORT_BY_NEWEST' | translate }}
              </li>
              <li
                nz-menu-item
                data-test-id="sort-queries-oldest"
                (click)="setQueriesSortBy('oldest')"
              >
                {{ 'COLLECTIONS_SORT_BY_OLDEST' | translate }}
              </li>
            </ul>
          </li>
        </ul>
      </nz-dropdown-menu>
    </div>
  </div>
  <div class="query-collections__item-content-wrapper" [ngClass]="{ 'query-collections__item-content--hidden': !showContent }">
    <div class="query-collections__item-nested">
      <app-query-collection-item
        *ngFor="let subcollectionTree of collectionTree.collections; trackBy:trackById;"
        [collectionTree]="subcollectionTree"
        [loggedIn]="loggedIn"
        (selectQueryChange)="selectQueryChange.next($event)"
        (deleteQueryChange)="deleteQueryChange.next($event)"
        (deleteCollectionChange)="deleteCollectionChange.next($event)"
        (editCollectionChange)="editCollectionChange.next($event)"
        (exportCollectionChange)="exportCollectionChange.next($event)"
      ></app-query-collection-item>
    </div>
    <div class="query-collections__item-queries">
      <div class="query-collections__item-queries-item" *ngFor="let query of sortedCollectionQueries(collectionTree.queries, sortBy); trackBy:trackById;">
        <span
          class="query-collections__item-queries-item-title"
          data-test-id="query-item-title"
          (click)="selectQueryChange.next({ query: query, collectionId: collectionTree.id, windowIdInCollection: query.id })"
        >
          <span class="query-title-icon"><app-icon name="file"></app-icon></span>
          {{ query.windowName }}
          <span
            *ngIf="!query.id"
            class="query-title-icon"
            [popper]="deprecatedCollectionQuery"
            [popperPlacement]="'bottom'"
          >
            <app-icon name="alert-triangle"></app-icon>
          </span>
          <popper-content #deprecatedCollectionQuery>
              {{ 'COLLECTIONS_DEPRECATED_QUERY_WARNING' | translate }}
              <a target="_blank" href="https://altairgraphql.dev/docs/tips/deprecated-collection-queries" rel="noopener">{{ 'LEARN_MORE_TEXT' | translate }}</a>
          </popper-content>
        </span>
        <span
          *ngIf="loggedIn && query.storageType === 'firestore'"
          class="query-collections__item-queries-item-synced-icon"
          [title]="'CLOUD_SYNC_TEXT' | translate"
        >
          <app-icon name="cloud"></app-icon>
        </span>
        <span class="query-collections__item-queries-item-right">
          <span class="query-collections__item-queries-item-icon">
            <button
              type="button"
              nz-dropdown
              nzTrigger="click"
              [nzDropdownMenu]="collectionQueryItemMenu"
              class="query-collections__item-icon__button"
            >
              <app-icon name="more-horizontal"></app-icon>
            </button>
          </span>
          <nz-dropdown-menu #collectionQueryItemMenu="nzDropdownMenu">
            <ul nz-menu>
              <li
                nz-menu-item
                data-test-id="delete-query"
                (click)="deleteQuery(query)"
              >
                {{ 'COLLECTIONS_QUERY_MENU_DELETE' | translate }}
              </li>
            </ul>
          </nz-dropdown-menu>
        </span>
      </div>
    </div>
  </div>
</div>
