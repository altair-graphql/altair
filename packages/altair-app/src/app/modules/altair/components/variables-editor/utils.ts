import {
  getNullableType,
  GraphQLInputObjectType,
  GraphQLInputType,
  isEnumType,
  isLeafType,
  isListType,
  isNullableType,
  isSpecifiedScalarType,
} from 'graphql';
import {
  JSONSchema7,
  JSONSchema7Definition,
  JSONSchema7Type,
  JSONSchema7TypeName,
} from 'json-schema';

/**
 * Maximum depth to traverse the GraphQL type tree for variable type resolution.
 * This is to prevent slow performance because of resolving the variable types and/or infinite recursion.
 */
const MAX_DEPTH = 4;

export type VariableToType = Record<string, GraphQLInputType>;

const specifiedScalarTypeToJSONSchemaType: Record<string, JSONSchema7TypeName> =
  {
    String: 'string',
    Int: 'integer',
    Float: 'number',
    Boolean: 'boolean',
    ID: 'string',
  };
const wrapWithAnyOfNull = (def: JSONSchema7, wrap?: boolean) => {
  if (!wrap) {
    return def;
  }
  return {
    anyOf: [def, { type: 'null' as JSONSchema7TypeName }],
  };
};
export const graphqlInputTypeToJsonSchemaType = (
  originalType: GraphQLInputType,
  depth = 0,
  defaultValue?: JSONSchema7Type
): JSONSchema7 => {
  if (depth > MAX_DEPTH) {
    // Max depth reached. Default to string
    return { type: 'string' };
  }

  const type = getNullableType(originalType);

  // Handle enums
  if (isEnumType(type)) {
    return wrapWithAnyOfNull(
      {
        type: 'string',
        description: type.description ?? undefined,
        default: defaultValue,
        anyOf: type.getValues().map((v) => ({
          enum: [v.value],
          description: v.description ?? undefined,
        })),
      },
      isNullableType(originalType)
    );
  }
  if (isLeafType(type)) {
    if (isSpecifiedScalarType(type)) {
      return wrapWithAnyOfNull(
        {
          type:
            specifiedScalarTypeToJSONSchemaType[type.toString()] ?? 'string',
          description: type.description ?? undefined,
          default: defaultValue,
        },
        isNullableType(originalType)
      );
    }
    // custom scalar types default to string
    return wrapWithAnyOfNull(
      {
        type: 'string',
        description: type.description ?? undefined,
      },
      isNullableType(originalType)
    );
  }
  // Handle lists
  if (isListType(type)) {
    return wrapWithAnyOfNull(
      {
        type: 'array',
        items: graphqlInputTypeToJsonSchemaType(type.ofType, depth + 1),
        description: type.ofType?.description ?? undefined,
        default: defaultValue,
      },
      isNullableType(originalType)
    );
  }

  if (type instanceof GraphQLInputObjectType) {
    return {
      type: 'object',
      description: type.description ?? undefined,
      properties: Object.entries(type.getFields()).reduce(
        (acc, [key, field]) => {
          acc[key] = graphqlInputTypeToJsonSchemaType(
            field.type,
            depth + 1,
            field.defaultValue
          );
          return acc;
        },
        {} as Record<string, JSONSchema7Definition>
      ),
    };
  }

  // Unknown case. Default to string
  return { type: 'string' };
};

/*
 * Converts a VariableToType object to a JSON Schema object.
 * This is used to generate a JSON Schema for the variables editor.
 */
export const vttToJsonSchema = (vtt?: VariableToType): JSONSchema7 => {
  if (!vtt) {
    return {
      type: 'object',
      properties: {},
    };
  }
  return {
    type: 'object',
    properties: Object.entries(vtt).reduce((acc, [key, type]) => {
      acc[key] = graphqlInputTypeToJsonSchemaType(type);
      return acc;
    }, {} as Record<string, JSONSchema7Definition>),
  };
};
