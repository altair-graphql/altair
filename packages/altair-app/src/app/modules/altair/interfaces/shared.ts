export interface IDictionary<V = any> {
  [key: string]: V;
}

export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;

export type TODO = any;

export type UnknownError = TODO;

export interface TrackByIdItem {
  id: string;
}
