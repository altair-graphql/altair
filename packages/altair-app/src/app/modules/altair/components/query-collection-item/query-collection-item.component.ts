import {
  Component,
  Input,
  Output,
  EventEmitter,
  ChangeDetectionStrategy,
} from '@angular/core';
import {
  IQuery,
  IQueryCollection,
  IQueryCollectionTree,
  SortByOptions,
} from 'altair-graphql-core/build/types/state/collection.interfaces';
import { NzModalService } from 'ng-zorro-antd/modal';
import { memoize } from '../../utils/memoize';

@Component({
  selector: 'app-query-collection-item',
  templateUrl: './query-collection-item.component.html',
  styleUrls: ['./query-collection-item.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class QueryCollectionItemComponent {
  @Input() collectionTree?: IQueryCollectionTree;
  @Input() loggedIn = false;
  @Input() queriesSortBy: SortByOptions = 'newest';

  @Output() selectQueryChange = new EventEmitter();
  @Output() deleteQueryChange: EventEmitter<{
    collectionId: string;
    query: IQuery;
  }> = new EventEmitter();
  @Output() deleteCollectionChange: EventEmitter<{
    collectionId: string;
  }> = new EventEmitter();
  @Output() editCollectionChange: EventEmitter<{
    collection: IQueryCollectionTree;
  }> = new EventEmitter();
  @Output() syncCollectionChange: EventEmitter<{
    collection: IQueryCollectionTree;
  }> = new EventEmitter();
  @Output() exportCollectionChange = new EventEmitter();
  @Output() sortCollectionQueriesChange = new EventEmitter<SortByOptions>();
  @Output() showQueryRevisionsChange = new EventEmitter<string>();
  @Output() copyQueryShareLinkChange = new EventEmitter<string>();

  showContent = true;

  constructor(private modal: NzModalService) {}

  getQueryCount(collection: IQueryCollectionTree) {
    return collection.queries && collection.queries.length;
  }

  toggleContent() {
    this.showContent = !this.showContent;
  }

  deleteQuery(query: IQuery) {
    this.modal.confirm({
      nzTitle:
        'Are you sure you want to delete this query from the collection?',
      nzContent: 'Note: This is an irreversible action.',
      nzOkText: 'Yes',
      nzCancelText: 'Cancel',
      nzOnOk: () => {
        if (this.collectionTree) {
          this.deleteQueryChange.next({
            query,
            collectionId: this.collectionTree.id,
          });
        }
      },
    });
  }

  showQueryRevisionsDialog(query: IQuery) {
    this.showQueryRevisionsChange.next(query.id);
  }

  deleteCollection() {
    this.modal.confirm({
      nzTitle: 'Are you sure you want to delete this collection?',
      nzContent: 'Note: This is an irreversible action.',
      nzOkText: 'Yes',
      nzCancelText: 'Cancel',
      nzOnOk: () => {
        if (this.collectionTree) {
          this.deleteCollectionChange.next({
            collectionId: this.collectionTree.id,
          });
        }
      },
    });
  }

  editCollection() {
    if (!this.collectionTree) {
      throw new Error('should never happen');
    }
    this.editCollectionChange.next({ collection: this.collectionTree });
  }

  syncCollection() {
    if (!this.collectionTree) {
      throw new Error('should never happen');
    }
    this.syncCollectionChange.next({ collection: this.collectionTree });
  }

  exportCollection() {
    if (!this.collectionTree) {
      throw new Error('should never happen');
    }
    this.exportCollectionChange.next({
      collectionId: this.collectionTree.id,
    });
  }
  setQueriesSortBy(sortBy: SortByOptions) {
    this.sortCollectionQueriesChange.emit(sortBy);
  }

  @memoize()
  sortedCollectionQueries(queries: IQuery[], sortBy: SortByOptions) {
    if (!queries) {
      return [];
    }

    switch (sortBy) {
      case 'a-z':
        return queries.sort((a, b) => {
          const aName = a.windowName.toLowerCase() || a.updated_at || '';
          const bName = b.windowName.toLowerCase() || b.updated_at || '';

          if (aName > bName) {
            return 1;
          }
          if (aName < bName) {
            return -1;
          }
          return 0;
        });
      case 'z-a':
        return queries.sort((a, b) => {
          const aName = a.windowName.toLowerCase() || a.updated_at || '';
          const bName = b.windowName.toLowerCase() || b.updated_at || '';

          if (aName > bName) {
            return -1;
          }
          if (aName < bName) {
            return 1;
          }
          return 0;
        });
      case 'oldest':
        return queries.sort((a, b) => {
          const aTimeStamp = a.updated_at || a.windowName.toLowerCase();
          const bTimeStamp = b.updated_at || b.windowName.toLowerCase();

          if (aTimeStamp > bTimeStamp) {
            return -1;
          }
          if (aTimeStamp < bTimeStamp) {
            return 1;
          }
          return 0;
        });
      case 'newest':
        return queries.sort((a, b) => {
          const aTimeStamp = a.updated_at || a.windowName.toLowerCase();
          const bTimeStamp = b.updated_at || b.windowName.toLowerCase();

          if (aTimeStamp > bTimeStamp) {
            return 1;
          }
          if (aTimeStamp < bTimeStamp) {
            return -1;
          }
          return 0;
        });
      default:
        return queries;
    }
  }

  trackById<T extends { id?: string }>(index: number, item: T) {
    return item.id;
  }
}
