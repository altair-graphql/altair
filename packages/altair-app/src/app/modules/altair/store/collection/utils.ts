import {
  CollectionState,
  IQueryCollection,
} from 'altair-graphql-core/build/types/state/collection.interfaces';
import { PerWindowState } from 'altair-graphql-core/build/types/state/per-window.interfaces';

export const COLLECTION_PATH_SEPARATOR = '/';

export const getCollections = (state: CollectionState) => state.list || [];
export const getCollection = (
  collections: IQueryCollection[],
  collectionId: string
) => collections.find((collection) => collection.id === collectionId);

export const getParentCollectionId = (collection: IQueryCollection) => {
  const id = collection.parentPath?.split(COLLECTION_PATH_SEPARATOR)?.pop();
  return id ? id : undefined;
};

export const getParentCollection = (
  collections: IQueryCollection[],
  collection: IQueryCollection
) => {
  const parentCollectionId = getParentCollectionId(collection);
  if (parentCollectionId) {
    return getCollection(collections, parentCollectionId);
  }
};

export const getAllParentCollections = (
  collections: IQueryCollection[],
  collection: IQueryCollection
) => {
  const result: IQueryCollection[] = [];
  let curCollection = collection;
  for (;;) {
    const parentCollection = getParentCollection(collections, curCollection);
    if (!parentCollection) {
      return result.reverse();
    }

    result.push(parentCollection);
    curCollection = parentCollection;
  }
};

export const getWindowParentCollections = (
  window: PerWindowState,
  collections: IQueryCollection[]
) => {
  if (window?.layout.windowIdInCollection && window?.layout.collectionId) {
    const collection = getCollection(collections, window.layout.collectionId);
    if (collection) {
      return [...getAllParentCollections(collections, collection), collection];
    }
  }
  return [];
};
