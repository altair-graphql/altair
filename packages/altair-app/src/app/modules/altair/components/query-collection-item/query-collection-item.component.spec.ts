import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';

import { QueryCollectionItemComponent } from './query-collection-item.component';
import { FormsModule } from '@angular/forms';
import { CodemirrorModule } from '@ctrl/ngx-codemirror';
import { TranslateModule } from '@ngx-translate/core';
import { NgxPopperModule } from 'ngx-popper';
import { SharedModule } from '../../modules/shared/shared.module';
import { NgxTestWrapper } from '../../../../../testing/wrapper';
import { mount } from '../../../../../testing/utils';
import { MockModule, MockProvider, ngMocks } from 'ng-mocks';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { expect } from '@jest/globals';
import { mock } from 'testing';
import { NzModalService } from 'ng-zorro-antd/modal';
import { IQueryCollectionTree } from 'altair-graphql-core/build/types/state/collection.interfaces';
ngMocks.globalExclude(NzModalService);
describe('QueryCollectionItemComponent', () => {
  let wrapper: NgxTestWrapper<QueryCollectionItemComponent>;
  const collectionData: IQueryCollectionTree = {
    id: '1',
    title: 'collection-1',
    collections: [],
    queries: [
      {
        id: 'query1',
        windowName: 'Query 1',
        updated_at: 1594249795629,
        type: 'window',
        apiUrl: '',
        headers: [],
        query: '',
        subscriptionUrl: '',
        variables: '',
        version: 1,
      },
      {
        id: 'query2',
        windowName: 'Another Query',
        updated_at: 1594249790029,
        type: 'window',
        apiUrl: '',
        headers: [],
        query: '',
        subscriptionUrl: '',
        variables: '',
        version: 1,
      },
      {
        id: 'query3',
        windowName: 'Zap Query',
        updated_at: 1594249790629,
        type: 'window',
        apiUrl: '',
        headers: [],
        query: '',
        subscriptionUrl: '',
        variables: '',
        version: 1,
      },
    ],
  };

  beforeEach(async () => {
    wrapper = await mount({
      component: QueryCollectionItemComponent,
      imports: [MockModule(SharedModule)],
      providers: [
        {
          provide: NzModalService,
          useValue: mock<NzModalService>({
            confirm({ nzOnOk } = {}) {
              if (nzOnOk) {
                (nzOnOk as any)();
              }
              return true as any;
            },
          }),
        },
      ],
      schemas: [NO_ERRORS_SCHEMA],
      propsData: {
        collectionTree: collectionData,
        queriesSortBy: 'newest',
      },
    });
  });

  it('should create', () => {
    expect(wrapper.component).toBeTruthy();
  });

  it('should render queries correctly', () => {
    expect(wrapper.component.nativeElement).toMatchSnapshot();
  });

  it('should emit "selectQueryChange" with query info when query is clicked', () => {
    const queryItemTitle = wrapper.find('[data-test-id="query-item-title"]');
    queryItemTitle.emit('click');

    expect(wrapper.emitted('selectQueryChange')).toBeTruthy();
    const emittedObj = wrapper.emitted('selectQueryChange')![0][0];
    expect(emittedObj).toEqual({
      query: {
        id: 'query2',
        windowName: 'Another Query',
        updated_at: 1594249790029,
        type: 'window',
        apiUrl: '',
        headers: [],
        query: '',
        subscriptionUrl: '',
        variables: '',
        version: 1,
      },
      windowIdInCollection: 'query2',
      collectionId: '1',
    });
  });

  it('should emit "editCollectionChange" when edit collection is clicked', () => {
    const editCollectionButton = wrapper.find(
      '[data-test-id="edit-collection"]'
    );
    editCollectionButton.emit('click');

    expect(wrapper.emitted('editCollectionChange')).toBeTruthy();
    const emittedObj = wrapper.emitted('editCollectionChange')![0][0];
    expect(emittedObj).toEqual({
      collection: collectionData,
    });
  });

  it('should emit "exportCollectionChange" when export collection is clicked', () => {
    const exportCollectionButton = wrapper.find(
      '[data-test-id="export-collection"]'
    );
    exportCollectionButton.emit('click');

    expect(wrapper.emitted('exportCollectionChange')).toBeTruthy();
    const emittedObj = wrapper.emitted('exportCollectionChange')![0][0];
    expect(emittedObj).toEqual({
      collectionId: '1',
    });
  });

  it('should emit "deleteCollectionChange" when delete collection is clicked', () => {
    const deleteCollectionButton = wrapper.find(
      '[data-test-id="delete-collection"]'
    );
    deleteCollectionButton.emit('click');

    expect(wrapper.emitted('deleteCollectionChange')).toBeTruthy();
    const emittedObj = wrapper.emitted('deleteCollectionChange')![0][0];
    expect(emittedObj).toEqual({
      collectionId: '1',
    });
  });

  it('should emit "deleteQueryChange" when delete query is clicked', () => {
    const deleteQueryButton = wrapper.find('[data-test-id="delete-query"]');
    deleteQueryButton.emit('click');

    expect(wrapper.emitted('deleteQueryChange')).toBeTruthy();
    const emittedObj = wrapper.emitted('deleteQueryChange')![0][0];
    expect(emittedObj).toEqual({
      collectionId: '1',
      query: collectionData.queries[0],
    });
  });

  it('should sort rendered queries when sortBy (a-z) is clicked', async () => {
    const sortByButton = wrapper.find('[data-test-id="sort-queries-a-z"]');
    sortByButton.emit('click');

    await wrapper.nextTick();

    wrapper.emitted('sortCollectionQueriesChange');
    const emittedObj = wrapper.emitted('sortCollectionQueriesChange')![0][0];
    expect(emittedObj).toEqual('a-z');
  });

  it('should render collapsed collection correctly', async () => {
    const collapseButton = wrapper.find('[data-test-id="collection-collapse"]');
    collapseButton.emit('click');

    await wrapper.nextTick();

    expect(wrapper.element).toMatchSnapshot();
  });
});
