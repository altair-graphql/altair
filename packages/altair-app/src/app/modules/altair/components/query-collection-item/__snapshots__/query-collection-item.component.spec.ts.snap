// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryCollectionItemComponent should render collapsed collection correctly 1`] = `
<app-query-collection-item>
  <div
    class="query-collections__item"
  >
    <div
      class="query-collections__item-details-wrapper"
    >
      <div
        class="query-collections__item-details"
        data-test-id="collection-collapse"
      >
        <div
          class="query-collections__item-title"
        >
          
          
          <span
            class="query-title-icon"
          >
            <app-icon
              name="folder-plus"
            />
          </span>
          <span>
            collection-1
          </span>
        </div>
        <div
          class="query-collections__item-meta"
        >
           3
        </div>
      </div>
      
      <div
        class="query-collections__item-right"
      >
        <span
          class="query-collections__item-icon"
        >
          <button
            class="query-collections__item-icon__button"
            nz-dropdown=""
            nztrigger="click"
            type="button"
          >
            <app-icon
              name="more-horizontal"
            />
          </button>
        </span>
        <nz-dropdown-menu>
          <ul
            nz-menu=""
          >
            
            <li
              data-test-id="edit-collection"
              nz-menu-item=""
            >
                
            </li>
            <li
              data-test-id="delete-collection"
              nz-menu-item=""
            >
                
            </li>
            <li
              data-test-id="export-collection"
              nz-menu-item=""
            >
                
            </li>
            <li
              nz-submenu=""
            >
              <ul>
                <li
                  data-test-id="sort-queries-a-z"
                  nz-menu-item=""
                >
                    
                </li>
                <li
                  data-test-id="sort-queries-z-a"
                  nz-menu-item=""
                >
                    
                </li>
                <li
                  data-test-id="sort-queries-newest"
                  nz-menu-item=""
                >
                    
                </li>
                <li
                  data-test-id="sort-queries-oldest"
                  nz-menu-item=""
                >
                    
                </li>
              </ul>
            </li>
          </ul>
        </nz-dropdown-menu>
      </div>
    </div>
    <div
      class="query-collections__item-content-wrapper query-collections__item-content--hidden"
    >
      <div
        class="query-collections__item-nested"
      >
        
      </div>
      <div
        class="query-collections__item-queries"
      >
        
        <div
          class="query-collections__item-queries-item"
        >
          <span
            class="query-collections__item-queries-item-title"
            data-test-id="query-item-title"
          >
            <span
              class="query-title-icon"
            >
              <app-icon
                name="file"
              />
            </span>
             Another Query 
            
            <popper-content>
                
              <a
                href="https://altair.sirmuel.design/docs/tips/deprecated-collection-queries"
                rel="noopener"
                target="_blank"
              >
                
              </a>
            </popper-content>
          </span>
          
          <span
            class="query-collections__item-queries-item-right"
          >
            <span
              class="query-collections__item-queries-item-icon"
            >
              <button
                class="query-collections__item-icon__button"
                nz-dropdown=""
                nztrigger="click"
                type="button"
              >
                <app-icon
                  name="more-horizontal"
                />
              </button>
            </span>
            <nz-dropdown-menu>
              <ul
                nz-menu=""
              >
                <li
                  data-test-id="delete-query"
                  nz-menu-item=""
                >
                    
                </li>
              </ul>
            </nz-dropdown-menu>
          </span>
        </div>
        <div
          class="query-collections__item-queries-item"
        >
          <span
            class="query-collections__item-queries-item-title"
            data-test-id="query-item-title"
          >
            <span
              class="query-title-icon"
            >
              <app-icon
                name="file"
              />
            </span>
             Zap Query 
            
            <popper-content>
                
              <a
                href="https://altair.sirmuel.design/docs/tips/deprecated-collection-queries"
                rel="noopener"
                target="_blank"
              >
                
              </a>
            </popper-content>
          </span>
          
          <span
            class="query-collections__item-queries-item-right"
          >
            <span
              class="query-collections__item-queries-item-icon"
            >
              <button
                class="query-collections__item-icon__button"
                nz-dropdown=""
                nztrigger="click"
                type="button"
              >
                <app-icon
                  name="more-horizontal"
                />
              </button>
            </span>
            <nz-dropdown-menu>
              <ul
                nz-menu=""
              >
                <li
                  data-test-id="delete-query"
                  nz-menu-item=""
                >
                    
                </li>
              </ul>
            </nz-dropdown-menu>
          </span>
        </div>
        <div
          class="query-collections__item-queries-item"
        >
          <span
            class="query-collections__item-queries-item-title"
            data-test-id="query-item-title"
          >
            <span
              class="query-title-icon"
            >
              <app-icon
                name="file"
              />
            </span>
             Query 1 
            
            <popper-content>
                
              <a
                href="https://altair.sirmuel.design/docs/tips/deprecated-collection-queries"
                rel="noopener"
                target="_blank"
              >
                
              </a>
            </popper-content>
          </span>
          
          <span
            class="query-collections__item-queries-item-right"
          >
            <span
              class="query-collections__item-queries-item-icon"
            >
              <button
                class="query-collections__item-icon__button"
                nz-dropdown=""
                nztrigger="click"
                type="button"
              >
                <app-icon
                  name="more-horizontal"
                />
              </button>
            </span>
            <nz-dropdown-menu>
              <ul
                nz-menu=""
              >
                <li
                  data-test-id="delete-query"
                  nz-menu-item=""
                >
                    
                </li>
              </ul>
            </nz-dropdown-menu>
          </span>
        </div>
      </div>
    </div>
  </div>
</app-query-collection-item>
`;

exports[`QueryCollectionItemComponent should render queries correctly 1`] = `
<app-query-collection-item>
  <div
    class="query-collections__item"
  >
    <div
      class="query-collections__item-details-wrapper"
    >
      <div
        class="query-collections__item-details"
        data-test-id="collection-collapse"
      >
        <div
          class="query-collections__item-title"
        >
          
          <span
            class="query-title-icon"
          >
            <app-icon
              name="folder-minus"
            />
          </span>
          
          <span>
            collection-1
          </span>
        </div>
        <div
          class="query-collections__item-meta"
        >
           3
        </div>
      </div>
      
      <div
        class="query-collections__item-right"
      >
        <span
          class="query-collections__item-icon"
        >
          <button
            class="query-collections__item-icon__button"
            nz-dropdown=""
            nztrigger="click"
            type="button"
          >
            <app-icon
              name="more-horizontal"
            />
          </button>
        </span>
        <nz-dropdown-menu>
          <ul
            nz-menu=""
          >
            
            <li
              data-test-id="edit-collection"
              nz-menu-item=""
            >
                
            </li>
            <li
              data-test-id="delete-collection"
              nz-menu-item=""
            >
                
            </li>
            <li
              data-test-id="export-collection"
              nz-menu-item=""
            >
                
            </li>
            <li
              nz-submenu=""
            >
              <ul>
                <li
                  data-test-id="sort-queries-a-z"
                  nz-menu-item=""
                >
                    
                </li>
                <li
                  data-test-id="sort-queries-z-a"
                  nz-menu-item=""
                >
                    
                </li>
                <li
                  data-test-id="sort-queries-newest"
                  nz-menu-item=""
                >
                    
                </li>
                <li
                  data-test-id="sort-queries-oldest"
                  nz-menu-item=""
                >
                    
                </li>
              </ul>
            </li>
          </ul>
        </nz-dropdown-menu>
      </div>
    </div>
    <div
      class="query-collections__item-content-wrapper"
    >
      <div
        class="query-collections__item-nested"
      >
        
      </div>
      <div
        class="query-collections__item-queries"
      >
        
        <div
          class="query-collections__item-queries-item"
        >
          <span
            class="query-collections__item-queries-item-title"
            data-test-id="query-item-title"
          >
            <span
              class="query-title-icon"
            >
              <app-icon
                name="file"
              />
            </span>
             Another Query 
            
            <popper-content>
                
              <a
                href="https://altair.sirmuel.design/docs/tips/deprecated-collection-queries"
                rel="noopener"
                target="_blank"
              >
                
              </a>
            </popper-content>
          </span>
          
          <span
            class="query-collections__item-queries-item-right"
          >
            <span
              class="query-collections__item-queries-item-icon"
            >
              <button
                class="query-collections__item-icon__button"
                nz-dropdown=""
                nztrigger="click"
                type="button"
              >
                <app-icon
                  name="more-horizontal"
                />
              </button>
            </span>
            <nz-dropdown-menu>
              <ul
                nz-menu=""
              >
                <li
                  data-test-id="delete-query"
                  nz-menu-item=""
                >
                    
                </li>
              </ul>
            </nz-dropdown-menu>
          </span>
        </div>
        <div
          class="query-collections__item-queries-item"
        >
          <span
            class="query-collections__item-queries-item-title"
            data-test-id="query-item-title"
          >
            <span
              class="query-title-icon"
            >
              <app-icon
                name="file"
              />
            </span>
             Zap Query 
            
            <popper-content>
                
              <a
                href="https://altair.sirmuel.design/docs/tips/deprecated-collection-queries"
                rel="noopener"
                target="_blank"
              >
                
              </a>
            </popper-content>
          </span>
          
          <span
            class="query-collections__item-queries-item-right"
          >
            <span
              class="query-collections__item-queries-item-icon"
            >
              <button
                class="query-collections__item-icon__button"
                nz-dropdown=""
                nztrigger="click"
                type="button"
              >
                <app-icon
                  name="more-horizontal"
                />
              </button>
            </span>
            <nz-dropdown-menu>
              <ul
                nz-menu=""
              >
                <li
                  data-test-id="delete-query"
                  nz-menu-item=""
                >
                    
                </li>
              </ul>
            </nz-dropdown-menu>
          </span>
        </div>
        <div
          class="query-collections__item-queries-item"
        >
          <span
            class="query-collections__item-queries-item-title"
            data-test-id="query-item-title"
          >
            <span
              class="query-title-icon"
            >
              <app-icon
                name="file"
              />
            </span>
             Query 1 
            
            <popper-content>
                
              <a
                href="https://altair.sirmuel.design/docs/tips/deprecated-collection-queries"
                rel="noopener"
                target="_blank"
              >
                
              </a>
            </popper-content>
          </span>
          
          <span
            class="query-collections__item-queries-item-right"
          >
            <span
              class="query-collections__item-queries-item-icon"
            >
              <button
                class="query-collections__item-icon__button"
                nz-dropdown=""
                nztrigger="click"
                type="button"
              >
                <app-icon
                  name="more-horizontal"
                />
              </button>
            </span>
            <nz-dropdown-menu>
              <ul
                nz-menu=""
              >
                <li
                  data-test-id="delete-query"
                  nz-menu-item=""
                >
                    
                </li>
              </ul>
            </nz-dropdown-menu>
          </span>
        </div>
      </div>
    </div>
  </div>
</app-query-collection-item>
`;
